go/teams/proofs_test.go

```diff
@@ -0,0 +1,84 @@
+package teams
+
+import (
+	"encoding/hex"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func createProofTerm(idInt int, seqno keybase1.Seqno) proofTerm {
+	var id [16]byte
+	id[15] = 0x25
+	id[0] = byte(idInt)
+	return proofTerm{
+		leafID: keybase1.UserOrTeamID(hex.EncodeToString(id[:])),
+		sigMeta: keybase1.SignatureMetadata{
+			SigChainLocation: keybase1.SigChainLocation{
+				Seqno: seqno,
+			},
+		},
+	}
+}
+
+func proofEq(x, y proof) bool {
+	return termEq(x.a, y.a) && termEq(x.b, y.b)
+}
+
+func termEq(a, b proofTerm) bool {
+	return a.leafID == b.leafID && a.sigMeta.SigChainLocation.Seqno == b.sigMeta.SigChainLocation.Seqno
+}
+
+func TestAddNeededHappensBeforeProof(t *testing.T) {
```

Comment:
I did the crossword puzzle and it seems like there are wires crossed. `e` before `f` and `k` before `l` is impossible?